ctru-rs/examples/file-explorer.rs

```diff
@@ -74,8 +75,37 @@ impl<'a> FileExplorer<'a> {
     }
 
     fn print_menu(&mut self) {
-        println!("Viewing {}", self.path.display());
+        match std::fs::metadata(&self.path) {
```

Comment:
I see this error after reading `romfs:/test-file.txt`:
```
Failed to read romfs:: Illegal byte sequence (os error 138)
```

This is probably due to the weird path handling? Maybe we should handle this case in the code until the underlying path issue is fixed?

Reply:
Yes, this is because of #13 as far as I know. I didn't want it to block this PR because I think it was an issue before but got a more exposed by this PR.

Reply:
Yeah, not blocking, but maybe we can do something like in `run` with `self.path.components().count() > 1` to avoid popping the path in this case. Or maybe we just need to reset it manually. Up to you.